frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -121,39 +225,341 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
-          </Trans>
-        </p>
-        {removePlugins && (
-          <p>
-            {t('olm~The console plugin provided by this operator will be disabled and removed.', {
-              count: enabledPlugins.length,
-            })}
-          </p>
-        )}
-        {uninstallMessage && (
+        {!isSubmitFinished ? (
           <>
-            <h2>{t('olm~Message from Operator developer')}</h2>
-            <p>{uninstallMessage}</p>
+            <Instructions
+              name={name}
+              namespace={namespace}
+              removePlugins={removePlugins}
+              enabledPlugins={enabledPlugins}
+              uninstallMessage={uninstallMessage}
+            />
+            <OperandsSection
+              operands={operands}
+              operandsLoaded={operandsLoaded}
+              operandsLoadedErrorMessage={operandsLoadedErrorMessage}
+              csvName={csvName}
+              cancel={cancel}
+              isDeleteAllOperandsSelected={isDeleteAllOperandsSelected}
+              setIsDeleteAllOperandsSelected={setIsDeleteAllOperandsSelected}
+            />
           </>
+        ) : (
+          <SubmitResults
+            operatorUninstallErrorMessage={operatorUninstallErrorMessage}
+            name={name}
+            namespace={namespace}
+            isDeleteAllOperandsSelected={isDeleteAllOperandsSelected}
+            operandDeletionErrors={operandDeletionErrors}
+            csvName={csvName}
+            cancel={cancel}
+          />
         )}
       </ModalBody>
       <ModalSubmitFooter
-        inProgress={inProgress}
-        errorMessage={errorMessage}
+        inProgress={isSubmitInProgress}
         cancel={cancel}
-        submitDanger
-        submitText={t('olm~Uninstall')}
+        submitDanger={!isSubmitFinished} // if submit finished show a non-danger 'OK'
+        submitText={t(isSubmitFinished ? 'olm~OK' : 'olm~Uninstall')}
       />
     </form>
   );
 };
 
+const Instructions: React.FC<InstructionProps> = ({
+  name,
+  namespace,
+  removePlugins,
+  enabledPlugins,
+  uninstallMessage,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <>
+      <p>
+        <Trans t={t} ns="olm">
+          Operator <strong>{{ name }}</strong> will be removed from <strong>{{ namespace }}</strong>
+          . Click on the checkbox below to also remove all Operands associated with this Operator.
+          If your Operator configured off-cluster resources, these will continue to run and require
+          manual cleanup.
+        </Trans>
+      </p>
+      {removePlugins && (
+        <p>
+          {t('olm~The console plugin provided by this operator will be disabled and removed.', {
+            count: enabledPlugins.length,
+          })}
+        </p>
+      )}
+      {uninstallMessage && (
+        <>
+          <h2>{t('olm~Message from Operator developer')}</h2>
+          <p>{uninstallMessage}</p>
+        </>
+      )}
```

Comment:
If we're going to pull these messages out into a separate component, I would make three components, one for each message.

Reply:
Hi @spadgett. I moved the `Instructions` inline, within the parent component, no parameters are being passed.  IMO, all these messages are 'instructions' which reside in this one 'component'.